### DISCUSSION TITLE
Support wakapi as another wakatime backend

### DISCUSSION THREAD

**Discussion by gaocegege** [lowlighter/metrics#563] [Ideas]
https://github.com/muety/wakapi is open-sourced wakatime backend implementation. It will be better if we can support it.

The only blocker here I can see is https://github.com/lowlighter/metrics/blob/master/source/plugins/wakatime/index.mjs#L22

```js
    const {data:{data:stats}} = await imports.axios.get(`${url}/api/v1/users/${user}/stats/${range}?api_key=${token}`)
```

The wakapi API is `/api/compat/wakatime/v1/`

Can we remove /api from the URL like this `${url}/v1/users...`?

Then wakatime users can use:

```
    plugin_wakatime_url: https://wakatime.com/api                 # Wakatime url endpoint
```

wakapi users can use:

```
    plugin_wakatime_url: https://wakapi.dev/api/compat/wakatime
```

**Comment by muety** [comment]
Looks like this is already supported: https://github.com/lowlighter/metrics/blob/fe74ec2b7a04edccdc9e0b55dc51e734cd4d8641/source/plugins/wakatime/metadata.yml#L57.

**Comment by lowlighter** [comment]
I have not tested but if you just want to remove `/api` because the api is located elsewhere, maybe `https://wakapi.dev/api/compat/wakatime/..` would do it? 

Granted it's kind of a hack but it should "remove" the forced `/api` in the plugin

**Comment by gaocegege** [comment]
@muety Seems that it is not related to route.

I used https://wakapi.dev/ and it returns 403 requested time range too broad. Because wakapi does not support query parameter `token` I think.

The request URL is `/api/v1/users/gaocegege/stats/last_7_days?api_key=<token>`

**Reply by gaocegege** [reply]
It works well after this PR. I am not sure if we can do it better.

https://github.com/gaocegege/wakapi/commit/630090e38a6c697fc6202c69e61d8467565115cd

**Reply by lowlighter** [reply]
Nice 👍

**Reply by gaocegege** [reply]
The PR is merged, I think we can close the discussion, thanks for your help!
